<a href="https://colab.research.google.com/github/Mvaleta54/Mvaleta54/blob/main/Percentage_of_users_that_use_Business.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##What is a percentage of our user's that travel for business purposes?

### Load the data

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('https://s3.eu-west-1.amazonaws.com/data.public.bdatainstitute.com/dam/python/booking.csv')
df

event_timestamp event_name visit_id  user_id      type  hotel_id
0       2019-01-04 00:06:35    booking  01973ce   393224   leisure      3888
1       2019-01-04 00:05:04    booking  060ba0b    85204   leisure      6283
2       2019-01-04 00:12:10    booking  030b1d5   141511   leisure      3273
3       2019-01-04 00:11:00    booking  0671f01   352800   leisure      3810
4       2019-01-04 00:17:39    booking  074b640   405479   leisure       331
...                     ...        ...      ...      ...       ...       ...
229741  2019-12-30 23:49:51    booking  0567a1c   482812   leisure      2932
229742  2019-12-30 23:48:06    booking  0427615   242892   leisure      5143
229743  2019-12-30 23:52:05    booking  02d8710   294790   leisure      3289
229744  2019-12-30 23:53:32    booking  04b7afd   365680   leisure      5279
229745  2019-12-31 00:03:34    booking  01c4cc3   333049  business      1440

[229746 rows x 6 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 229746 entries, 0 to 229745
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   event_timestamp  229746 non-null  object
 1   event_name       229746 non-null  object
 2   visit_id         229746 non-null  object
 3   user_id          229746 non-null  int64 
 4   type             229746 non-null  object
 5   hotel_id         229746 non-null  int64 
dtypes: int64(2), object(4)
memory usage: 10.5+ MB


### First approach - Subset

In [8]:
df.type == 'business'

0         False
1         False
2         False
3         False
4         False
          ...  
229741    False
229742    False
229743    False
229744    False
229745     True
Name: type, Length: 229746, dtype: bool

In [9]:
df[df.type == 'business']

event_timestamp event_name visit_id  user_id      type  hotel_id
23      2019-01-04 01:20:00    booking  051de09   197917  business      6363
103     2019-01-04 09:54:28    booking  10ea45e   501589  business      5673
109     2019-01-04 10:23:41    booking  04b40b8   418723  business       346
236     2019-01-04 13:32:14    booking  05c2990   524978  business       275
261     2019-01-04 14:09:01    booking  0632139   422527  business      4626
...                     ...        ...      ...      ...       ...       ...
229606  2019-12-30 18:28:18    booking  0784f1b   397797  business      4917
229607  2019-12-30 18:26:55    booking  074a1c5   146944  business      6769
229666  2019-12-30 20:13:18    booking  02f251a   220242  business      6962
229683  2019-12-30 21:08:29    booking  08b5b04   184488  business      1443
229745  2019-12-31 00:03:34    booking  01c4cc3   333049  business      1440

[7843 rows x 6 columns]

In [10]:
df.head(5)

event_timestamp event_name visit_id  user_id     type  hotel_id
0  2019-01-04 00:06:35    booking  01973ce   393224  leisure      3888
1  2019-01-04 00:05:04    booking  060ba0b    85204  leisure      6283
2  2019-01-04 00:12:10    booking  030b1d5   141511  leisure      3273
3  2019-01-04 00:11:00    booking  0671f01   352800  leisure      3810
4  2019-01-04 00:17:39    booking  074b640   405479  leisure       331

In [11]:
df[df.type == 'business'].count()

event_timestamp    7843
event_name         7843
visit_id           7843
user_id            7843
type               7843
hotel_id           7843
dtype: int64

In [13]:
sum(df.type == 'business')

7843

In [14]:
len(df)

229746

In [15]:
sum(df.type == 'business') / len(df)

0.034137699894666286

Our answer would be around 3.41% business travels.

### Is this the correct answer? 

In [24]:
df[df.type == 'business'].user_id

23        197917
103       501589
109       418723
236       524978
261       422527
           ...  
229606    397797
229607    146944
229666    220242
229683    184488
229745    333049
Name: user_id, Length: 7843, dtype: int64

In [25]:
df[df.type == 'business'].user_id.nunique()

7776

In [26]:
df.user_id.nunique()

184190

In [27]:
df[df.type == 'business'].user_id.nunique() / df.user_id.nunique()

0.04221727563928552

Our answer would be around 4.2% business travels.

### Second approach - group by

In [20]:
df.groupby('type').user_id.nunique()

type
business      7776
leisure     179139
Name: user_id, dtype: int64

In [21]:
sum(df.groupby('type').user_id.nunique())

186915

In [22]:
df.groupby('type').user_id.nunique() / sum(df.groupby('type').user_id.nunique())

type
business    0.041602
leisure     0.958398
Name: user_id, dtype: float64

Our answer would be around 4.1% business travels and 95.8% leisure travels.

In [28]:
df.groupby('type').user_id.nunique() / df.user_id.nunique()

type
business    0.042217
leisure     0.972577
Name: user_id, dtype: float64

### Which is correct?